demo/frosting/build/BuildSetup.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using Cake.Common.Diagnostics;
+using Cake.Core;
+using Cake.Frosting;
+
+namespace Build
+{
+    public class BuildSetup : IFrostingSetup
+    {
+        public void Setup(ICakeContext context)
+        {
+            var ctx = context as BuildContext ?? throw new ArgumentException("Not my context!");
```

Comment:
Doesn't Frosting allow for generic `Setup<T>` so you don't have to do this type cast to `BuildContext `?

Reply:
Yes: [FrostingSetup\<TContext\>](https://cakebuild.net/api/Cake.Frosting/FrostingSetup_1/)

Reply:
In my defense: I did search.. (Not sure if that makes it better or worse, though.)